### PR TITLE
feat: allow user config MaxConnsPerHost (#90)

### PR DESCRIPTION
**Pull request by wolfbolin**
This is just a draft. I think there still needs to be a discussion about how the Config class is written.

### CODE REVIEW

**Comment by wolfbolin** [comment]
I would like to add a `NetworkConfig` class, like `AuthConfig` , for configuring TLS, GZIP, number of connections, etc. However, the change causes the existing configuration to be unavailable.

Is this a reasonable and acceptable change?

**Comment by shoothzj** [comment]
First, I agree to make `MaxConnsPerHost` configurable.
Second, we haven't release `1.0.0` version yet, any reasonable change is acceptable.
But, I still have concerns about `NetworkConfig`'s name, `gunzip` in NetworkConfig seems werid. I was thinking about `TransportConfig`, but we only support http now, and host, port is in the top field(Config). `HttpConfig` also feels werid.

**Comment by shoothzj** [comment]
Let's add `MaxConnsPerHost` in to configuration first to solve #90.
For configration refactor, I think we can open a new thread.

File: opengemini/client_impl.go [hunk 1]
```diff
@@ -89,24 +89,14 @@ func buildEndpoints(addresses []*Address, tlsEnabled bool) []endpoint {
...
-	} else {
-		return &http.Client{
-			Timeout: config.Timeout,
-			Transport: &http.Transport{
-				DialContext: (&net.Dialer{
-					Timeout: config.ConnectTimeout,
-				}).DialContext,
-			},
-		}
+	return &http.Client{
```
Comment (shoothzj) [inline opengemini/client_impl.go:92]: Nice patch! the if clause is no-needed

**Review by shoothzj — APPROVED** [review]

**Pull request merged**